jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -801,7 +807,7 @@ public Response payAllInvoices(@PathParam("accountId") final UUID accountId,
             final BigDecimal amountToPay = (remainingRequestPayment.compareTo(invoice.getBalance()) >= 0) ?
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
-                final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
+                final UUID paymentMethodId = externalPayment ? account.getPaymentMethodId() : inputPaymentMethodId;
```

Comment:
Actually, this part is wrong -- my bad!! This became obvious as i reviewed the current diff: When `externalPayment` is null, we pass a null `paymentMethodId` and the flag `externalPayment` to the api. So you should have the following instead:

```
UUID paymentMethodId = null;
if (!externalPayment) {
    paymentMethodId = inputPaymentMethodId != null ? inputPaymentMethodId : account.getPaymentMethodId();
}
```

OR if using ternary forms all the way:

```
final UUID paymentMethodId = externalPayment ?
                             null :
                             (inputPaymentMethodId != null ? inputPaymentMethodId : account.getPaymentMethodId());
```